src/udev/udev-rules-util.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "escape.h"
+#include "macro.h"
+#include "utf8.h"
+
+static inline int parse_value(char *str, char **ret_value, char **ret_endpos) {
+        char *value = str;
+        char *i, *j;
+        int r;
+        bool is_escaped;
+
+        /* value must be double quotated */
+        if (value[0] == '"') {
+                is_escaped = false;
+                value++;
+        } else if (value[0] == 'e') {
+                if (value[1] != '"')
+                        return -EINVAL;
+                is_escaped = true;
+                value += 2;
+        } else {
+                return -EINVAL;
+        }
+
+        /* unescape double quotation '\"' -> '"' */
+        for (i = j = value;; i++, j++) {
+                if (*i == '"')
+                        break;
+                if (*i == '\0')
+                        return -EINVAL;
+                if (i[0] == '\\' && i[1] == '"') {
+                        if (is_escaped) {
+                                *j = '\\';
+                                j++;
+                        }
+                        i++;
+                }
+                *j = *i;
+        }
+        j[0] = '\0';
+
+        if (is_escaped) {
+                char *unescaped;
```

Comment:
Please use `_cleanup_free_` attribute, and add a blank line after the variable declaration:
```
_cleanup_free_ char *unescaped = NULL;

r = cunescape_lenght(...);
```